enterprise/backend/test/metabase_enterprise/caching/api_test.clj

```diff
@@ -0,0 +1,109 @@
+                                   :next-value     {:strategy "nocache" :config {:name "root"}}}}
+                        (last-audit-event)))))
+
+            (testing "Can configure others"
+              (is (mt/user-http-request :crowberto :put 200 "ee/caching/"
+                                        {:model    "database"
+                                         :model_id (:id db)
+                                         :strategy {:type "nocache" :name "db"}}))
+              (is (mt/user-http-request :crowberto :put 200 "ee/caching/"
+                                        {:model    "dashboard"
+                                         :model_id (:id dash1)
+                                         :strategy {:type "nocache" :name "dash"}}))
+              (is (mt/user-http-request :crowberto :put 200 "ee/caching/"
+                                        {:model    "question"
+                                         :model_id (:id card1)
+                                         :strategy {:type "nocache" :name "card1"}})))
+
+            (testing "HTTP responds with correct listings"
+              (is (=? {:items [{:model "root" :model_id 0}]}
+                      (mt/user-http-request :crowberto :get 200 "ee/caching/")))
+              (is (=? {:items [{:model "database" :model_id (:id db)}]}
+                      (mt/user-http-request :crowberto :get 200 "ee/caching/" {}
+                                            :model :database)))
+              (is (=? {:items [{:model "dashboard" :model_id (:id dash1)}
+                               {:model "question" :model_id (:id card1)}]}
+                      (mt/user-http-request :crowberto :get 200 "ee/caching/" {}
+                                            :collection (:id col1) :model :dashboard :model :question))))
+
+            (testing "We select correct config for something from a db"
+              (testing "First card has own config"
+                (is (=? {:type :nocache :name "card1"}
+                        (:cache-strategy (#'qp.card/query-for-card card1 {} {} {} {}))))
+                (is (=? {:type :nocache :name "card1"}
+                        (:cache-strategy (#'qp.card/query-for-card card1 {} {} {} {:dashboard-id (u/the-id dash1)})))))
+              (testing "Second card should hit database or dashboard cache"
+                (is (=? {:type :nocache :name "db"}
+                        (:cache-strategy (#'qp.card/query-for-card card2 {} {} {} {}))))
+                (is (=? {:type :nocache :name "dash"}
+                        (:cache-strategy (#'qp.card/query-for-card card2 {} {} {} {:dashboard-id (u/the-id dash1)})))))
+              (testing "Third card targets other db and gets root config"
+                (is (=? {:type :nocache :name "root"}
+                        (:cache-strategy (#'qp.card/query-for-card card3 {} {} {} {}))))))
+
+            (testing "It's possible to delete a configuration"
```

Comment:
We might as well test that when we GET the listing the config is gone there too.

Reply:
Agree, done in b3d0e32fcf